google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/DirectExecuteResultSetTest.java

```diff
@@ -0,0 +1,242 @@
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class DirectExecuteResultSetTest {
+
+  private DirectExecuteResultSet createSubject() {
+    ResultSet delegate =
+        ResultSets.forRows(
+            Type.struct(StructField.of("test", Type.int64())),
+            Arrays.asList(Struct.newBuilder().set("test").to(1L).build()));
+    return DirectExecuteResultSet.ofResultSet(delegate);
+  }
+
+  @Test
+  public void testMethodCallBeforeNext()
+      throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+    List<String> excludedMethods =
+        Arrays.asList("getStats", "next", "close", "ofResultSet", "equals", "hashCode");
+    DirectExecuteResultSet subject = createSubject();
+    callMethods(subject, excludedMethods, IllegalStateException.class);
+  }
+
+  @Test
+  public void testMethodCallAfterClose()
+      throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+    List<String> excludedMethods =
+        Arrays.asList(
+            "getStats",
+            "next",
+            "close",
+            "getType",
+            "getColumnCount",
+            "getColumnIndex",
+            "getColumnType",
+            "ofResultSet",
+            "equals",
+            "hashCode");
+    DirectExecuteResultSet subject = createSubject();
+    subject.next();
+    subject.close();
+    callMethods(subject, excludedMethods, IllegalStateException.class);
+  }
+
+  @Test
+  public void testMethodCallAfterNextHasReturnedFalse()
+      throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+    List<String> excludedMethods =
+        Arrays.asList(
+            "getStats",
+            "next",
+            "close",
+            "getType",
+            "getColumnCount",
+            "getColumnIndex",
+            "getColumnType",
+            "ofResultSet",
+            "equals",
+            "hashCode");
+    DirectExecuteResultSet subject = createSubject();
+    subject.next();
+    subject.next();
+    callMethods(subject, excludedMethods, IndexOutOfBoundsException.class);
+  }
+
+  private void callMethods(
+      DirectExecuteResultSet subject,
+      List<String> excludedMethods,
+      Class<? extends Exception> expectedException)
+      throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+    for (Method method : DirectExecuteResultSet.class.getDeclaredMethods()) {
+      if (Modifier.isPublic(method.getModifiers()) && !excludedMethods.contains(method.getName())) {
+        boolean exception = false;
+        try {
+          if (method.getParameterTypes().length == 0) {
+            method.invoke(subject);
+          } else if (method.getParameterTypes().length == 1
```

Comment:
minor - for readability, can we extract some of these out into descriptive variables?

Reply:
Number of parameters and parameter type have been extracted to variables and the if-statement has been partially re-written as a switch statement.